core-runner/src/runner.rs

```diff
@@ -351,7 +352,7 @@ impl<MQ: MessageQueue, PS: ProgramStorage, WL: WaitList, E: Environment<Ext>>
             self.block_height,
         );
 
-        if run_result.outcome.was_trap() {
+        if run_result.outcome.was_trap() && next_message_exit_code != 1 {
```

Comment:
Why are you checking that the exit code is not equal to 1? Usually zero is a successful completion and everything else is an error code